packages/sourcecred/src/cli/update/utils.js

```diff
@@ -0,0 +1,24 @@
+// @flow
+
+import cloneDeep from "lodash.clonedeep";
+
+// This is a type-friendly way to add optional attributes to a config.
+// Attributes from the `optionals` parameter will only be added if they
+// are not undefined.
+// It is recommended to define T explicitly when using this.
+// Example: `const object = buildObject<MyType>(...);`
+export function buildObject<T: {}>(
+  required: T,
+  optionals: $Shape<T>,
+  exclusions: $ReadOnlyArray<$Values<T> | typeof undefined> = [undefined]
+): T {
+  let config: T = cloneDeep(required);
```

Comment:
Nit: drop this extra dependency ref and just do this:
```suggestion
  let config: T = {...required};
```

Reply:
But if there are objects nested within `required` then those objects will still be passed by reference. I think it's better to have a clean deep clone.